### ISSUE TITLE
Moodle, SAML2, ADFS 4

### ISSUE THREAD

**Issue by darrenkharris** [catalyst/moodle-auth_saml2#137]
We have setup the SAML2 plugin to work with our ADFS 4 install.  As part of this setup we have entered the public IdP metadata xml link and also created a relying trust in ADFS by entering the SP xml file link.  
When we run both test settings under the Moodle plugin page we get this result after being redirected to and signing in through our Federated Services page:

Authed!

array(2) {
  ["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn"]=>
  array(1) {
    [0]=>
    string(20) "staff.test@nua.ac.uk"
  }
  ["http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname"]=>
  array(1) {
    [0]=>
    string(10) "staff.test"
  } 

We have the Display IdP link on our sign in page and when we attempt to sign in using this we get the screen as shown below:

![image](https://user-images.githubusercontent.com/31772018/30220490-7c93f164-94b7-11e7-9ca1-d94d074c217e.png)

We have tried various IdP mappings but whatever we change the mapping to, the end result is the same message "You have logged in succesfully but we could not find your 'sAMAccountName' attribute to associate you to an account in Moodle." except 'sAMAccountName' changes to whatever we have put in the IdP Mapping field.

Can anybody help identify what the cause and resolution of this issue is?

Many thanks

Darren

**Comment by brendanheywood** [comment]
Do you want to auto create users on demand when they auth? Is so then:

![image](https://user-images.githubusercontent.com/187449/30301146-3f4c99f4-979c-11e7-9246-5b2061062a90.png)

If you don't want to create users on demand then you need to prevision them ahead of time using some other means, like manual csv import, or ldap sync etc and the mdlattr needs to match whatever saml attribute you need it to.

**Comment by darrenkharris** [comment]
Thanks for the reply.  
At the moment we do have our users auto created in Moodle via an LDAP sync and when we try to sign in with one of the test users where we have set the authentication type to SAML we get the error.  
I did out of curiousity change the auto create user setting over the weekend and tred it but still got the same message although I was trying to sign in with a user that had already been created via the LDAP synch.
I have changed the mdlattr to the Username and tried this with no luck then EMailAddress and again no joy.
Users that are created via LDAP sync can login fine however when the authentication type is set to SAML they can't.
